theories/distributions/Distr.ec

```diff
@@ -1489,6 +1576,98 @@ move=> le_qp; rewrite drestrictE &(mu_eq).
 by move=> x @/predI; apply/eq_iff/andb_idl/le_qp.
 qed.
 
+(* -------------------------------------------------------------------- *)
+
+op dcond (d : 'a distr) (p : 'a -> bool) = dscale (drestrict d p).
```

Comment:
This is more a question than a comment. Currently, in the case of subdistributions, conditioning on the output satisfying some predicate implies conditioning on there being some output in the first place. I suppose this is what we want, but maybe we should add a comment here.